sunspot/lib/sunspot/util.rb

```diff
@@ -188,6 +188,29 @@ def escape(value)
         RSolr.solr_escape(value).gsub(/([\s\.])/, '\\\\\1')
       end
 
+      def parse_json_facet(field_name, options, setup)
+        field = setup.field(field_name)
+        if options[:time_range]
+          unless field.type.is_a?(Sunspot::Type::TimeType)
+            raise(
+              ArgumentError,
+              ':time_range can only be specified for Date or Time fields'
+            )
+          end
+          Sunspot::Query::DateFieldJsonFacet.new(field, options, setup)
+        elsif options[:range]
+          unless [Sunspot::Type::TimeType, Sunspot::Type::FloatType, Sunspot::Type::IntegerType ].inject(false){|res,type| res || field.type.is_a?(type)}
```

Comment:
lets change this to `find`, `inject` seem redundant here

Reply:
I'v got that line from
sunspot/sunspot/lib/sunspot/dsl/field_query.rb:289

maybe is to change there too?